File: java-sdk/src/main/java/io/fyno/core_java/utils/Logger.java
package io.fyno.core_java.utils;

import android.util.Log;

public class Logger {
    private static LogLevel Level = LogLevel.VERBOSE;
    public static void setLevel(LogLevel newLevel) {

    }
    public static void i(String tag, String message) {
        if (Level.getNum() <= LogLevel.INFO.getNum()) {
            Log.i(tag, message);
        }
    }

    public static void w(String tag, String message) {
        if (Level.getNum() <= LogLevel.ERROR.getNum()) {
            Log.d(tag, message);
        }
    }

    public static void w(String tag, String message, Throwable throwable) {
        if (Level.getNum() <= LogLevel.ERROR.getNum()) {
            Log.d(tag, message, throwable);
        }
    }

    public static void e(String tag, String message) {
        if (Level.getNum() <= LogLevel.ERROR.getNum()) {
            Log.e(tag, message);
        }
    }

    public static void e(String tag, String message, Throwable throwable) {
        if (Level.getNum() <= LogLevel.ERROR.getNum()) {
            Log.e(tag, message, throwable);
        }
    }

    public static void d(String tag, String message) {
        if (Level.getNum() <= LogLevel.DEBUG.getNum()) {
            Log.d(tag, message);
        }
    }
}

File: fyno-kotlin-core/src/main/java/io/fyno/kotlin_sdk/utils/Logger.kt
package io.fyno.kotlin_sdk.utils

import android.util.Log

enum class LogLevel(val num: Int) {
    VERBOSE(1),
    DEBUG(2),
    INFO(3),
    ERROR(4),
    OFF(Int.MAX_VALUE)
}

object Logger {
    var Level = LogLevel.VERBOSE
    fun i(tag: String, message: String) {
        if (logAllowed(LogLevel.INFO.num))
            Log.i(tag, message)
    }
    fun e(tag: String, message: String, throwable: Throwable? = null) {
        if (logAllowed(LogLevel.ERROR.num))
            Log.e(tag, message, throwable)
    }
    private fun logAllowed(level: Int): Boolean {
        return Level.num <= level
    }
}
